sim/battle-stream.ts

```diff
@@ -220,6 +220,7 @@ export function getPlayerStreams(stream: BattleStream) {
 				streams.p1.push(p1Update);
 				const p2Update = data.replace(/\n\|split\n[^\n]*\n[^\n]*\n([^\n]*)\n[^\n]*/g, '\n$1').replace(/\n\n/g, '\n');
 				streams.p2.push(p2Update);
+				// TODO p3/p4?
```

Comment:
@MacChaeger shouldn't we be doing something with the `p3` and `p4` streams here?

Reply:
Per Zarel’s request, we’re still using a 4 channel system. `p3` and ``p4`` use channels 2 and 3.

Reply:
2 and 3 of 1-4 that is

Reply:
I'm not sure how that works? Your teammate gets to see your exact HP etc in a 2v2? and what about free for all?

Regardless of how the `'|split'` is supposed to work - we set up a `ObjectReadWriteStream` for `p3` and `p4` here and don't push to them in the `update` case (only `sideupdate`/`end` and errors). Is this intentional? From your explanation it seems like we should be pushing the `p1update` to `p3` and the `p2update` to `p4`? Or does `update` never get used in multi-battles, in which case we should at least add a comment

Reply:
Yes, your teammate gets to see exact HP in 4 player battles, even when your teammate isn't really your teammate, because Zarel didn't think that mattered enough to use 6 channels. That said, I guess we probably should be pushing the updates like you suggested, though I'm not entirely sure what this function even gets used for.

Reply:
Hmm, not sure I agree with that? I'm going to open a different PR for this because its unrelated to `floating-promises` and I'd like to reserve this PR for Zarel and I to keep writing technical essays back and forth at each other. ;)